quint/src/generated/Quint.g4

```diff
@@ -176,6 +174,12 @@ expr:           // apply a built-in operator via the dot notation
         |       '{' expr '}'                                        # braces
         ;
 
+// match e { A(a) => e1 | B => e2 | C(_) => e3 | ... | _ => en }
+matchSumExpr: MATCH expr '{' '|'? matchCase+=matchSumCase ('|' matchCase+=matchSumCase)* '}' ;
+matchSumCase: (variantMatch=matchSumVariant | wildCardMatch='_') '=>' expr ;
+matchSumVariant
+    : (variantLabel=simpleId["variant label"]) ('(' (variantParam=simpleId["match case parameter"] | '_') ')')? ;
```

Comment:
How are these labels used? I see they have spaces, so I suppose it's only for error reporting to users? If so, that's really great - we should really use them more.

Reply:
Yep, the string arguments are used for error reporting in

https://github.com/informalsystems/quint/blob/ba946ad88befae93404bef9fd88ac1a283451722/quint/src/generated/Quint.g4#L243-L254

To produce messages like

```
>>> {A::B::c: 3}
syntax error: error: [QNT008] Identifiers in a record cannot be qualified with '::'. Found A::B::c.
{A::B::c: 3}
        ^


>>> type T = A::B(int)
syntax error: error: [QNT008] Identifiers in a variant label cannot be qualified with '::'. Found A::B.
type T = A::B(int)
             ^
```

The ANTLR labels like `variantLabel` and `variantParam`, on the other hand, are used to provide named accessors for the parsed data on the rhs of the `=`, used in

https://github.com/informalsystems/quint/blob/ba946ad88befae93404bef9fd88ac1a283451722/quint/src/parsing/ToIrListener.ts#L996-L1003